base/compiler/compiler.jl

```diff
@@ -137,6 +137,9 @@ include("compiler/optimize.jl") # TODO: break this up further + extract utilitie
 include("compiler/bootstrap.jl")
 ccall(:jl_set_typeinf_func, Cvoid, (Any,), typeinf_ext_toplevel)
 
+const _COMPILER_WORLD_AGE = ccall(:jl_get_tls_world_age, UInt, ())
+compiler_world_age() = _COMPILER_WORLD_AGE
```

Comment:
Is this OK since `jl_typeinf_world` is serialized? Or should I add a C function `jl_get_typeinf_world`?

Reply:
I don't think this will work. You'll need to ask C what the world age of inference is.

Reply:
`jl_typeinf_world` is set by `jl_set_typeinf_func` just above. IIUC, this `jl_typeinf_world` is baked in the sysimage. So, isn't age correct unless `jl_set_typeinf_func` is called again?

Of course, I'm fine with adding a proper C function but I'm curious when/how it'd be wrong.

Reply:
I suppose you're right. World ages do persist across sysimage serialization (but not incremental serialization), so this should be ok.

Reply:
We could pontially replace `jl_call_in_typeinf_world` with this too. While normally preserving these values would be risky, this is Core.Compiler, and is loaded early enough it is often not too important, but `Revise` expects to be able to change this, so it might still be a bit of a nuisance.

So might be better to write this as:
```julia
function invoke_in_compiler(args...)
    return ccall(:jl_call_in_typeinf_world, Any, (Ptr{Ptr{Cvoid}}, Cint), Any[args...], length(args))
end
```